contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -201,41 +175,54 @@ contract OptimismPortal {
             _data
         );
 
-        // Verify proof that a withdrawal on L2 was initated
-        if (
+        // Verify that the hash of this withdrawal was stored in the withdrawal contract on L2. If
+        // this is true, then we know that this withdrawal was actually triggered on L2 can can
+        // therefore be relayed on L1.
+        require(
             WithdrawalVerifier._verifyWithdrawalInclusion(
                 withdrawalHash,
                 _outputRootProof.withdrawerStorageRoot,
                 _withdrawalProof
-            ) == false
-        ) {
-            revert InvalidWithdrawalInclusionProof();
-        }
+            ),
+            "OptimismPortal: invalid withdrawal inclusion proof"
+        );
 
-        // Check that this withdrawal has not already been finalized.
-        if (finalizedWithdrawals[withdrawalHash] == true) {
-            revert WithdrawalAlreadyFinalized();
-        }
+        // Check that this withdrawal has not already been finalized, this is replay protection.
+        require(
+            finalizedWithdrawals[withdrawalHash] == false,
+            "OptimismPortal: withdrawal has already been finalized"
+        );
 
-        // Set the withdrawal as finalized
+        // Mark the withdrawal as finalized so it can't be replayed.
         finalizedWithdrawals[withdrawalHash] = true;
 
-        // Save enough gas so that the call cannot use up all of the gas
-        require(gasleft() >= _gasLimit + 20000, "Insufficient gas to finalize withdrawal.");
+        // We want to maintain the property that the amount of gas supplied to the call to the
+        // target contract is at least the gas limit specified by the user. We can do this by
+        // enforcing that, at this point in time, we still have gaslimit + buffer gas available.
+        require(
+            gasleft() >= _gasLimit + 20000,
+            "OptimismPortal: insufficient gas to finalize withdrawal"
+        );
 
-        // Set the l2Sender so that other contracts can know which account
-        // on L2 is making the withdrawal
+        // Set the l2Sender so contracts know who triggered this withdrawal on L2.
         l2Sender = _sender;
-        // Make the call and ensure that a contract cannot out of gas
-        // us by returning a huge amount of data
+
+        // TODO: Do we want reentrancy protection by checking that l2Sender is not the default L2
```

Comment:
I think that handling withdrawals that call back into this function with a different withdrawal would be weird control flow. Is there a usecase? It would be something like the first withdraw calls out to a contract that then looks at `finalizedWithdrawals` and then sees the first withdrawal in there and then calls back in. Building the calldata for this would be extremely expensive and cannot be easily done on chain. I feel like its not a good usecase and we should add the reentrency protection just for safety purposes

Reply:
Ok I'll add it just to be safe